web/src/components/Calendar.js

```diff
@@ -0,0 +1,307 @@
+        ...theme.typography.body2,
+        display: 'inline-block',
+    },
+    title: {
+        ...theme.typography.h6,
+        color: theme.palette.text.secondary,
+        fontWeight: theme.typography.fontWeightBold,
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+    },
+    icon: {
+        color: theme.palette.action.active,
+        verticalAlign: 'middle',
+    },
+    circle: {
+        width: theme.spacing(4.5),
+        height: theme.spacing(4.5),
+        verticalAlign: 'super',
+    },
+    textCenter: {
+        textAlign: 'center',
+    },
+    dateAndTitle: {
+        color: '#78909C',
+        lineHeight: 1.1,
+    },
+    titleContainer: {
+        paddingBottom: theme.spacing(2),
+    },
+    container: {
+        paddingBottom: theme.spacing(1.5),
+    },
+});
+
+// #FOLD_BLOCK
+const CellBase = React.memo(
+    ({
+        classes,
+        startDate,
+        formatDate,
+        otherMonth,
+        // #FOLD_BLOCK
+    }) => {
+        const isFirstMonthDay = startDate.getDate() === 1;
+        const formatOptions = isFirstMonthDay
+            ? { day: 'numeric', month: 'long' }
+            : { day: 'numeric' };
+        return (
+            <TableCell
+                tabIndex={0}
+                className={classNames({
+                    [classes.cell]: true,
+                    [classes.opacity]: otherMonth,
+                })}>
+                <div className={classes.content}></div>
+                <div className={classes.text}>
+                    {formatDate(startDate, formatOptions)}
+                </div>
+            </TableCell>
+        );
+    }
+);
+
+const TimeTableCell = withStyles(styles, { name: 'Cell' })(CellBase);
+
+const Appointment = withStyles(styles, {
+    name: 'Appointment',
+})(({ classes, ...restProps }) => (
+    <Appointments.Appointment {...restProps} className={classes.appointment} />
+));
+
+const AppointmentContent = withStyles(styles, {
+    name: 'AppointmentContent',
+})(({ classes, ...restProps }) => (
+    <Appointments.AppointmentContent
+        {...restProps}
+        className={classes.apptContent}
+    />
+));
+
+const FlexibleSpace = withStyles(styles, { name: 'ToolbarRoot' })(
+    ({ classes, ...restProps }) => (
+        <Toolbar.FlexibleSpace {...restProps} className={classes.flexibleSpace}>
+            <div className={classes.flexContainer}>
+                <Typography variant="h5" style={{ marginLeft: '10px' }}>
+                    My Calendar
+                </Typography>
+            </div>
+        </Toolbar.FlexibleSpace>
+    )
+);
+
+const getRandomColor = () => {
+    var letters = '0123456789ABCDEF';
+    var color = '#';
+    for (var i = 0; i < 6; i++) {
+        color += letters[Math.floor(Math.random() * 16)];
+    }
+    return color;
+};
+let setColor = false;
+
+const Calendar = ({ appointments }) => {
+    const [data, setData] = useState([]);
+    const [owners, setOwners] = useState([]);
```

Comment:
we use the term 'provider' and 'buyer' throughout the project, we should stick to one of those instead of owner